lib/rule-context.js

```diff
@@ -72,42 +72,27 @@ var PASSTHROUGHS = [
  * @param {object} ecmaFeatures The ecmaFeatures settings passed from the config file.
  */
 function RuleContext(ruleId, eslint, severity, options, settings, ecmaFeatures) {
+    // public.
+    this.id = ruleId;
+    this.options = options;
+    this.settings = settings;
+    this.ecmaFeatures = ecmaFeatures;
+
+    // private.
+    this.eslint = eslint;
+    this.severity = severity;
```

Comment:
I tried to add `Object.freeze(this)` into the end of this constructor, then `npm run perf` (target is `lib`) come to slow about 60ms (2%).

Hmm, I will try to detect that with `RuleTester`.